### ISSUE TITLE
Factor _trial_temp directory creation and cleanup so as to be re-usable by multiple runner implementations

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4564]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4564|
|Type|enhancement|
|Created|2010-07-09 21:05:34Z|
|Branch|https://github.com/twisted/twisted/tree/trial-temp-reusage-4564|

`TrialRunner` has various logic for selecting a _trial_temp[-N] directory, cleaning up what was there before, creating a new one, and changing the current working directory to be it.

This is probably generally useful to runner implementations.  As it happens, [#1784](https://github.com/twisted/twisted/issues/1784) (disttrial) introduces a new runner implementation which benefits from being able to use this code.

<details><summary>Searchable metadata</summary>

```
trac-id__4564 4564
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_trial_temp_reusage_4564 branches/trial-temp-reusage-4564
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1278709534000000 1278709534000000
changetime__1279287947000000 1279287947000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29751]) Branching to 'trial-temp-reusage-4564'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Pulled the code out of the disttrial branch, cleaned it up a bit.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [29757]) Drop this refactoring; see [#4564](https://github.com/twisted/twisted/issues/4564).

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Screwtape set owner to Screwtape<br>Screwtape set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|Screwtape set owner to @exarkun<br>Screwtape set status to `new`|
|-|-|

Looks good to me, go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [29851]) Merge trial-temp-reusage-4564 

Author: exarkun
Reviewer: screwtape
Fixes: [#4564](https://github.com/twisted/twisted/issues/4564)

Factor `_trial_temp` creation and cleanup code out of the runner so it can
be re-used by another runner implementation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
